packages/api/utils/requestValidation.ts

```diff
@@ -0,0 +1,25 @@
+import { NO_SUCH_PORTFOLIO_DRAFT, REQUEST_BODY_INVALID, REQUEST_BODY_EMPTY } from "./errors";
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { SetupError, SetupResult, SetupSuccess } from "./response";
+import { isBodyPresent, isValidJson, isValidUuidV4 } from "./validation";
+
+/**
+ * Check if incoming POST Request passes basic shape validation
+ *
+ * @param event - The incoming API Gateway Request proxied to Lambda
+ * @returns SetUpSuccess object if event passes validation, otherwise it returns SetUpError
+ */
+export function postRequestShapeValidation<T>(event: APIGatewayProxyEvent): SetupResult<T> {
+  if (!isValidUuidV4(event.pathParameters?.portfolioDraftId)) {
+    return new SetupError(NO_SUCH_PORTFOLIO_DRAFT);
+  }
+  if (!isBodyPresent(event.body)) {
+    return new SetupError(REQUEST_BODY_EMPTY);
+  }
+  const portfolioDraftId = event.pathParameters!.portfolioDraftId!;
```

Comment:
I know we know why this is safe but I'd like to get into the habit of documenting why we know any use if `!` is safe. Notice how we were able to avoid it for `body`? Something about how if `pathParameters` _isn't_ present, line 13 would have caught it and we'd have returned already anyway (of course, don't say line 13, describe it).